pkg/deploy/plugin_registry.go

```diff
@@ -27,49 +27,93 @@ type PluginRegistryConfigMap struct {
 }
 
 const (
-	PluginRegistry = "plugin-registry"
+	PluginRegistry              = "plugin-registry"
+	pluginRegistryGatewayConfig = "plugin-registry-gtw"
 )
 
 /**
  * Create plugin registry resources unless an external registry is used.
  */
-func SyncPluginRegistryToCluster(checluster *orgv1.CheCluster, clusterAPI ClusterAPI) (bool, error) {
+func SyncPluginRegistryToCluster(checluster *orgv1.CheCluster, cheHost string, clusterAPI ClusterAPI) (bool, error) {
 	pluginRegistryURL := checluster.Spec.Server.PluginRegistryUrl
 	if !checluster.Spec.Server.ExternalPluginRegistry {
 		var host string
+		exposureStrategy := util.GetServerExposureStrategy(checluster, DefaultServerExposureStrategy)
+		singleHostExposureType := util.GetSingleHostExposureType(checluster, DefaultKubernetesSingleHostExposureType, DefaultOpenShiftSingleHostExposureType)
+		useGateway := exposureStrategy == "single-host" && (util.IsOpenShift || singleHostExposureType == "gateway")
+
 		if !util.IsOpenShift {
-			ingress, err := SyncIngressToCluster(checluster, PluginRegistry, "", PluginRegistry, 8080, clusterAPI)
-			if !util.IsTestMode() {
-				if ingress == nil {
-					logrus.Infof("Waiting on ingress '%s' to be ready", PluginRegistry)
-					if err != nil {
-						logrus.Error(err)
+			if useGateway {
+				cfg := GetGatewayRouteConfig(checluster, pluginRegistryGatewayConfig, "/"+PluginRegistry, 1, "http://"+PluginRegistry+":8080")
```

Comment:
IMO registries and keycloak must have higher priority than che-server. I don't know what's the Treafik behavior with same priorities, so it may actually work, but I would keep it clear and set it to e.g. 10, to have prio 1 for che-server, 10 for registries and keycloak, 100 for workspaces